### PR TITLE
Monkeys can pop balloons with darts.

### PR DESCRIPTION
**Pull request by Wallemations**
## About The Pull Request
Lets monkeys pop balloons with foam darts. Bloons drop iron coins when popped this way.

## Why It's Good For The Game
bloons-monkey

## Changelog

:cl: Wallem
expansion: Monkeys can now protect their village from the rubber menace.
/:cl:

### CODE REVIEW

**Comment by Addust** [comment]
y e s

File: code/game/objects/items/toys.dm [hunk 1]
```diff
@@ -134,6 +134,19 @@
...
+		// Monkeys make money from popping bloons
+		pop_balloon(monkey_pop = TRUE)
+	else
+		return ..()
+
+/obj/item/toy/balloon/proc/pop_balloon(monkey_pop = FALSE)
+	playsound(src, 'sound/effects/cartoon_pop.ogg', 50, vary = TRUE)
+	if(monkey_pop)
+		new /obj/item/coin/iron(src)
+	qdel()
```
Comment (Mothblocks) [inline code/game/objects/items/toys.dm:148]: qdel...What

Comment (Wallemations) [reply code/game/objects/items/toys.dm:148]: I cannot believe I forgot that.

**Comment by Shadowflame909** [comment]
But where do you get the balloons from

File: code/game/objects/items/toys.dm (hunk 1 above)
Comment (Wallemations) [reply code/game/objects/items/toys.dm]: Fixed

**Comment by Wallemations** [comment]
> But where do you get the balloons from

Arcade machines

**Comment by Wallemations** [comment]
Hold on melbert told me a way I might be able to make it work when thrown too

**Review by tralezab — APPROVED** [review]

**Comment by CRITAWAKETS** [comment]
yo when can we upgrade the monke with spike ball catapults

**Comment by tralezab** [comment]
Code looks good 😋

**Comment by optimumtact** [comment]
Did you test this?

**Comment by Wallemations** [comment]
> Did you test this?

Yes I did several times. I was reminded why it's a good idea to test your code lmao.
It all worked when I tested it.

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -134,6 +134,31 @@
...
+		return ..()
+
+/obj/item/toy/balloon/hitby(atom/movable/AM, skipcatch, hitpush, blocked, datum/thrownthing/throwingdatum)
+	if(ismonkey(throwingdatum.thrower) && istype(AM, /obj/item/ammo_casing/caseless/foam_dart))
+		pop_balloon(monkey_pop = TRUE)
+	else
+		return ..()
+
+/obj/item/toy/balloon/bullet_act(obj/projectile/P)
+	if((istype(P,/obj/projectile/bullet/p50) || istype(P,/obj/projectile/bullet/reusable/foam_dart)) && ismonkey(P.firer))
```
Comment (ATH1909) [inline code/game/objects/items/toys.dm:150]: why only p50s instead of all bullets?

**Comment by ATH1909** [comment]
an AI thing to make npc monkeys do this on their own would be nice, but that's probably just bikeshedding

File: code/game/objects/items/toys.dm (hunk 2 above)
Comment (Wallemations) [reply code/game/objects/items/toys.dm:150]: Sniper monkeys.

When was the last time you saw a monkey with a Glock in Bloons

**Comment by Wallemations** [comment]
> an AI thing to make npc monkeys do this on their own would be nice, but that's probably just bikeshedding

That is far beyond my capabilities

File: code/game/objects/items/toys.dm (hunk 2 above)
Comment (Fryzzie) [reply code/game/objects/items/toys.dm:150]: In bloons td battles one, The cobra agent was a monkey with a glock

Comment (ATH1909) [reply code/game/objects/items/toys.dm:150]: and there's, like, a gatling gun monkey

Comment (Wallemations) [reply code/game/objects/items/toys.dm:150]: Is there a gatling gun in SS13??

**Pull request merged**